scripts/generate_dump

```diff
@@ -190,15 +190,13 @@ save_proc() {
 ###############################################################################
 # Dumps all fields and values from given Redis DB.
 # Arguments:
-#  DB id: id of DB for redis-cli
 #  DB name: filename to which output will be saved
 # Returns:
 #  None
 ###############################################################################
 save_redis() {
-    local db=$1
-    local db_name=$2
-    save_cmd "redis-dump -d $db -s /var/run/redis/redis.sock -y" "$db_name.json"
+    local db_name=$1
+    save_cmd "sonic-db-dump -n $db_name -y" "$db_name.json"
```

Comment:
>$db_name [](start = 31, length = 8)

Escape it to prevent input injection.

Reply:
is this what you mean to?

Reply:
Sorry I mean quotation instead of escaping the $db_name variable. If it is quoted rightly, sonic-db-dump will find malicious input easily.
No need to add new restriction on db_name here.

---
In reply to: [377958584](https://github.com/Azure/sonic-utilities/pull/810#discussion_r377958584) [](ancestors = 377958584)

Reply:
OK. GOT IT. UPDATED